answerking_app/tests/test_unit/test_unit_order_serializer.py

```diff
@@ -0,0 +1,469 @@
+from decimal import Decimal
+from typing import OrderedDict
+from unittest import mock
+
+from freezegun import freeze_time
+from rest_framework.utils.serializer_helpers import ReturnDict
+
+from answerking_app.models.models import Order, Product, LineItem
+from answerking_app.models.serializers import OrderSerializer
+from answerking_app.tests.test_unit.UnitTestBaseClass import UnitTestBase
+
+
+class OrderSerializerUnitTests(UnitTestBase):
+    UBT = UnitTestBase()
+    serializer_path: str = "answerking_app.models.serializers."
+    test_prod_1_data: dict = UBT.get_fixture(
+        "products", "plain_burger_data.json"
+    )
+    test_prod_2_data: dict = UBT.get_fixture(
+        "products", "margarita_pizza_data.json"
+    )
+    test_order_1_data: dict = UBT.get_fixture("orders", "order_data.json")
+    test_order_2_data: dict = UBT.get_fixture("orders", "order_data_2.json")
+
+    frozen_time: str = "2022-01-01T01:02:03.000000Z"
+    frozen_time_update: str = "2022-07-01T01:32:03.000000Z"
+
+    @freeze_time(frozen_time)
+    def setUp(self):
+        valid_order: Order = Order.objects.create()
+        product_burger: Product = Product.objects.create(
+            **self.test_prod_1_data
+        )
+        self.test_order_1_data["lineItems"][0]["product"][
+            "id"
+        ] = product_burger.id
+        product_pizza: Product = Product.objects.create(
+            **self.test_prod_2_data
+        )
+        self.test_order_2_data["lineItems"][0]["product"][
+            "id"
+        ] = product_pizza.id
+
+        line_item = LineItem.objects.create(
+            order=valid_order, product=product_burger, quantity=3
+        )
+        line_item.calculate_sub_total()
+        valid_order.calculate_total()
+
+    def tearDown(self):
+        Order.objects.all().delete()
+        Product.objects.all().delete()
+        LineItem.objects.all().delete()
+
+    def test_order_serializer_contains_correct_fields(self):
+        order: Order = Order.objects.first()
+        test_serializer_data: ReturnDict = OrderSerializer(order).data
+        expected: list[str] = [
+            "id",
+            "createdOn",
+            "lastUpdated",
+            "orderStatus",
+            "orderTotal",
+            "lineItems",
+        ]
+        actual: list[str] = list(test_serializer_data.keys())
+        self.assertEqual(actual, expected)
+
+    def test_order_serializer_id_field_content(self):
+        order: Order = Order.objects.first()
+        test_serializer_data: ReturnDict = OrderSerializer(order).data
+        expected: int = order.id
+        actual: int = test_serializer_data["id"]
+        self.assertEqual(actual, expected)
+
+    def test_order_serializer_products_details_content(self):
+        order: Order = Order.objects.first()
```

Comment:
would probably use objects.all() to get all orders and then test that only one order is returned then set order to orders[0]. See product serializer in dev for example. would do this method wherever you are doing objects.first(). this way we are confirming that we are getting the order we want.